packages/collector/test/tracing/cloud/aws/sqs/receiveMessage.js

```diff
@@ -0,0 +1,205 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+'use strict';
+
+const instana = require('../../../../../')();
+const express = require('express');
+const port = process.env.APP_RECEIVER_PORT || 3216;
+const agentPort = process.env.INSTANA_AGENT_PORT;
+const request = require('request-promise');
+
+const app = express();
+
+const delay = require('../../../../../../core/test/test_util/delay');
+const { sqs } = require('./sqsUtil');
+const queueURL = process.env.AWS_SQS_QUEUE_URL;
+const logPrefix = `AWS SQS Message Receiver (${process.pid}):\t`;
+
+let hasStartedPolling = false;
+
+const receiveParams = {
+  AttributeNames: ['SentTimestamp'],
+  MaxNumberOfMessages: 1,
+  MessageAttributeNames: ['All'],
+  QueueUrl: queueURL,
+  VisibilityTimeout: 5,
+  WaitTimeSeconds: 5
+};
+
+function log() {
+  /* eslint-disable no-console */
+  const args = Array.prototype.slice.call(arguments);
+  args[0] = `${logPrefix}${args[0]}`;
+  console.log.apply(console, args);
+}
+
+app.get('/', (_req, res) => {
+  if (hasStartedPolling) {
+    res.send('OK');
+  } else {
+    res.status(500).send('Not ready yet.');
+  }
+});
+
+app.listen(port, () => {
+  `Receiver server started at port ${port}`;
+});
+
+function receivePromise() {
+  let span;
+  return sqs
+    .receiveMessage(receiveParams)
+    .promise()
+    .then(data => {
+      if (!data || !data.Messages || data.Messages.length === 0) {
+        log('no messages, doing nothing');
+        return;
+      }
+
+      span = instana.currentSpan();
+      span.disableAutoEnd();
+
+      log('got messages:', data.Messages.map(m => m.MessageId));
+
+      const deleteParams = {
+        QueueUrl: queueURL,
+        ReceiptHandle: data.Messages[0].ReceiptHandle
+      };
+
+      return sqs
+        .deleteMessage(deleteParams)
+        .promise()
+        .then(() => delay(200))
+        .then(() => request(`http://127.0.0.1:${agentPort}`))
+        .then(() => {
+          log('The follow up request after receiving a message has happened.');
+          span.end();
+        })
+        .catch(err => {
```

Comment:
I guess we could omit this catch I guess and just have the outer then have a catch. Since we return this inner promise, an error in this inner promise should bubble up to the outer promise I think.